### PR TITLE
Fix adding embed frameworks script phase to unit test targets if xcframeworks are present.

### LINKED ISSUE #10652: Bug with test spec / static linkage / XCFramework - Failed to load the test bundle. The bundle is damaged or missing necessary resources. Library not loaded. Reason: image not found.
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

The unit tests (test spec) of a certain local/development pod fail at run time with the following error: 

Screenshot 2021-05-13 at 11 12 52

```
Assertions: System: Failed to load the test bundle. If you believe this error represents a bug, please attach the result bundle at /Users/okp/Library/Developer/Xcode/DerivedData/DemoApp-bmsiniphoqhqciawekumfhsuqwjj/Logs/Test/Test-DemoApp-2021.05.13_11-11-46-+0100.xcresult. (Underlying Error: The bundle “SomeModule-Unit-Tests” couldn’t be loaded because it is damaged or missing necessary resources. The bundle is damaged or missing necessary resources. Try reinstalling the bundle. dlopen_preflight(/Users/okp/Library/Developer/Xcode/DerivedData/DemoApp-bmsiniphoqhqciawekumfhsuqwjj/Build/Products/Debug-iphonesimulator/SomeModule-Unit-Tests.xctest/SomeModule-Unit-Tests): Library not loaded: @rpath/MUXSDKStats.framework/MUXSDKStats
  Referenced from: /Users/okp/Library/Developer/Xcode/DerivedData/DemoApp-bmsiniphoqhqciawekumfhsuqwjj/Build/Products/Debug-iphonesimulator/SomeModule-Unit-Tests.xctest/SomeModule-Unit-Tests
  Reason: image not found)
```

### The error seems to manifest because:
- The `Podfile` specifies `use_frameworks! :linkage => :static`
   _(if you change to `use_frameworks!` only, things work fine)_ 
   
- The development pod/module has an **XCFramework** dependency (`Mux-Stats-AVPlayer`)
   _(if you change `SomeModule`s dependency from `Mux-Stats-AVPlayer` to `RxSwift`, things work fine)_ 
   
_**Unfortunately I am not able to change either of these things in my reall-world project.**_

### Workarounds tried and failed:
I have tried the following, as suggested in other threads, but the outcome is exactly the same...
- `test_spec.requires_app_host = true`
- `"ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES" => "YES"`

## What did you do?

- Clone the (highly simplified) example project from https://github.com/OliverPearmain/CocoapodsTestBundleIssueExample
- Run `pod install`
- Open Xcode, select the `DemoApp` scheme and run the tests 
   _(note that the `DemoApp` scheme has `SomeModule-Unit-Tests` included in it's test suite)_

## What did you expect to happen?

✅ `SomeModule-Unit-Tests` run and pass
✅ `DemoAppTests` run and pass

## What happened instead?

❌ `SomeModule-Unit-Tests` do not run. They encounter the run time error mentioned above.
✅ `DemoAppTests` run and pass

### Stack

```
   CocoaPods : 1.10.1
        Ruby : ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-darwin19]
    RubyGems : 3.1.2
        Host : macOS 11.2.1 (20D75)
       Xcode : 12.4 (12D4e)
         Git : git version 2.28.0
Ruby lib dir : /Users/okp/.rbenv/versions/2.7.1/lib
Repositories : cocoapods - git - https://github.com/CocoaPods/Specs.git @ 2d42b44d22a8a238f6f0ff8b271e5c18678de571

               gitlab-mxx-podspecs - git - https://triller-iosdevs:d_5uy-uuZNrGeXbvuuT6@gitlab.com/mxx-ios-kits/mxx-podspecs.git @ ef7590b0aab9f97b2f2acca93bf465637e5a00ef

               trillerco - git - https://github.com/TrillerCo/private-pod-trunk.git @ caad05a7e0f23452a61e767f0eae58a89e4ee9e0

               trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /Users/okp/.rbenv/versions/2.7.1/bin/pod
```

### Plugins

```
cocoapods-binary       : 0.4.4
cocoapods-deintegrate  : 1.0.4
cocoapods-dependencies : 1.3.0
cocoapods-plugins      : 1.0.0
cocoapods-search       : 1.0.0
cocoapods-stats        : 1.1.0
cocoapods-trunk        : 1.5.0
cocoapods-try          : 1.2.0
```

### Podfile

```ruby
platform :ios, '12.0'

use_frameworks! :linkage => :static

target 'DemoApp' do

  pod 'SomeModule', :path => 'Modules/SomeModule', :testspecs => ['Tests']

  target 'DemoAppTests' do
  end

end
```

## Project that demonstrates the issue

https://github.com/OliverPearmain/CocoapodsTestBundleIssueExample

### PR DESCRIPTION
**Pull request by dnkoutso**
closes #10652 

integration specs https://github.com/CocoaPods/cocoapods-integration-specs/pull/320

### CODE REVIEW

File: spec/unit/installer/xcode/pods_project_generator/pod_target_integrator_spec.rb [hunk 1]
```diff
@@ -201,6 +201,28 @@ class PodTargetIntegrator
...
+                @watermelon_pod_target.stubs(:resource_paths).returns({})
+                @watermelon_pod_target.stubs(:xcframeworks).returns('WatermelonLib' => [Pod::Xcode::XCFramework.new('CoconutLib', fixture('CoconutLib.xcframework'))])
+                test_native_target = stub('TestNativeTarget', :symbol_type => :unit_test_bundle, :build_phases => [],
+                                                              :shell_script_build_phases => [], :project => @project,
+                                                              :name => 'WatermelonLib-Unit-Tests')
+                installation_result = TargetInstallationResult.new(@watermelon_pod_target, @native_target,
+                                                                   [], [test_native_target])
+                PodTargetIntegrator.new(installation_result, :use_input_output_paths => true).integrate!
+                test_native_target.build_phases.map(&:display_name).should == [
+                  '[CP] Embed Pods Frameworks',
```
Comment (dnkoutso) [inline spec/unit/installer/xcode/pods_project_generator/pod_target_integrator_spec.rb:216]: verified this and the checks below were failing without my change.

File: spec/unit/installer/xcode/pods_project_generator/pod_target_integrator_spec.rb [hunk 2]
```diff
@@ -201,6 +201,28 @@ class PodTargetIntegrator
                 ]
               end
 
+              it 'integrates xcframeworks, with input and output file lists even when frameworks are emptyb' do
```
Comment (paulb777) [inline spec/unit/installer/xcode/pods_project_generator/pod_target_integrator_spec.rb:204]: nit: "empty"

**Review by paulb777 — APPROVED** [review]

File: spec/unit/installer/xcode/pods_project_generator/pod_target_integrator_spec.rb (hunk 2 above)
Comment (dnkoutso) [reply spec/unit/installer/xcode/pods_project_generator/pod_target_integrator_spec.rb:204]: ah thanks will fix.

Comment (dnkoutso) [reply spec/unit/installer/xcode/pods_project_generator/pod_target_integrator_spec.rb]: fixed!

**Pull request merged**